osquery/tables/networking/windows/arp_cache.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+*  Copyright (c) 2014-present, Facebook, Inc.
+*  All rights reserved.
+*
+*  This source code is licensed under the BSD-style license found in the
+*  LICENSE file in the root directory of this source tree. An additional grant
+*  of patent rights can be found in the PATENTS file in the same directory.
+*
+*/
+
+#include <boost/range/algorithm/find.hpp>
+#include <osquery/tables.h>
+#include <string>
+
+#include "boost/algorithm/string/replace.hpp"
+#include "osquery/core/conversions.h"
+#include "osquery/core/windows/wmi.h"
+#include "osquery/tables/networking/windows/interfaces.h"
+
+namespace osquery {
+namespace tables {
+
+const std::map<unsigned short, const std::string> mapOfAddressFamily = {
+    {2, "IPv4"}, {23, "IPv6"},
+};
+
+const std::map<unsigned char, const std::string> mapOfStore = {
+    {0, "Persistent"}, {1, "Active"},
+};
+
+const std::map<unsigned char, const std::string> mapOfState = {
+    {0, "Unreachable"},
+    {1, "Incomplete"},
+    {2, "Probe"},
+    {3, "Delay"},
+    {4, "Stale"},
+    {5, "Reachable"},
+    {6, "Permanent"},
+    {7, "TBD"},
+};
+
+QueryData genArpCache(QueryContext& context) {
+  Row r;
+  QueryData results;
+  QueryData interfaces = genInterfaceDetails(context);
+  WmiRequest wmiSystemReq("select * from MSFT_NetNeighbor",
+                          L"ROOT\\StandardCimv2");
```

Comment:
Looks like when dealing with wide strings in VS, it's preferred to add a cast to `(BSTR)` in front. Checkout the conformance PR that's up: https://github.com/facebook/osquery/pull/2860/files